src/solver/conv_direct_naive_conv_bwd.cpp

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/solver.hpp>
+#include <miopen/conv/data_invoke_params.hpp>
+#include <miopen/conv/wrw_invoke_params.hpp>
+#include <miopen/env.hpp>
+#include <miopen/visit_float.hpp>
+#include <miopen/conv/conv_direct_naive_conv.hpp>
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CONV_DIRECT_NAIVE_CONV_BWD)
+
+namespace miopen {
+namespace solver {
+
+bool ConvDirectNaiveConvBwd::IsApplicable(const ConvolutionContext& ctx) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_DIRECT_NAIVE_CONV_BWD{}))
```

Comment:
This ENV switch should not be applicable for cases when DirectNaive is used to compute reference in the driver.

Right away, the simplest solution for this is:
- introducing a global boolean flag in the `miopen::debug` namespace, e.g. `AlwaysEnableConvDirectNaive`
- setting it to `true` in the driver on entry to `Run*GPUReference()` and resetting it prior exit
  - please use guard idiom for that; example -- search for `AutoMiopenWarmupMode` in the conv_driver.hpp.

Example of using of similar approach is `FindEnforceDisable`:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/4ae5799347d139c87553d62e6164d0cee630cb13/src/include/miopen/find_controls.hpp#L36-L43

The same is applicable to `ConvDirectNaiveConvFwd` and `ConvDirectNaiveConvWrw`

Reply:
This problem is a side effect of my previous comment, https://github.com/ROCmSoftwarePlatform/MIOpen/pull/583#discussion_r535570814. Hopefully, this debug switch is much simpler than an internal API.

Reply:
Oh this is to **always** use naive kernel when used as compute reference. Yes let me add this feature

Reply:
@atamazov I put the `AlwaysEnableConvDirectNaive` declaration in `src/include/miopen/solver.hpp`, then can be used in `driver/conv_driver.hpp`. Since I move the `conv_direct_naive_conv.hpp` to `src/solver/conv_direct_naive_conv.hpp`, and if put here is not a good choice to include into `driver/conv_driver.hpp`